### ISSUE TITLE
Order coordination

### ISSUE THREAD

**Issue by bohgman** [durinfab/WebApps-Assignment7#1]
Hallo Fabian und Tobias,

ich habe ein paar Fragen zu dem Auftrag.

Genügt für einen Drink oder eine Sorte Boba eine ID und ein Name oder sollen weitere Attribute zugeordnet werden?

Wie genau soll eine Bestellung funktionieren? Enthält ein Boba Tea genau einen Drink und eine Sorte Boba? Eine Bestellung hat dann folgende Attribute: ID, Getränk, Boba, Datum, Bewertung?

Die Kurzbeschreibung von der Bewertung ist identisch mit der zum Abrufen einer Bestellung. Ist das richtig so? Ist eine Bewertung im Zahlenbereich zwischen 0 und 5 sowie einer Anzeige bis auf eine Nachkommastelle ok?

Wie soll die Trennung zwischen Mitarbeiter und Kunde erfolgen? Genügt es die Funktionen zur Organisation von Drinks und Bobas über ein Passwort zu sichern? 

Gibt es ein oder zwei bevorzugte Farben für das Design der Website?

Beste Grüße

------------------------------------------------------------------------------------------------------------

Hello Fabian and Tobias,

I have a few questions about the assignment.

Is an ID and a name enough for a drink or a variety of Boba or should other attributes be assigned?

How exactly should an order work? Does a Boba Tea contain exactly one drink and one sort of Boba? An order then has the following attributes: ID, Drink, Boba, Date, Rating?

The short description of the rating is identical to the one used to retrieve an order. Is this correct? Is a rating in the number range between 0 and 5 and a display up to one decimal place ok?

How should the separation between employee and customer be done? Is it enough to secure the functions for organizing drinks and bobas via a password? 

Are there one or two preferred colors for the website design?

Best regards

**Comment by durinfab** [comment]
Hallo Manuel,

> Genügt für einen Drink oder eine Sorte Boba eine ID und ein Name oder sollen weitere Attribute zugeordnet werden?

Es genügt wenn jedes Liquid und jede Sorte Boba eine ID, einen Namen (welche den Geschmack repräsentiert) und eine Beschreibung besitzen.

> Wie genau soll eine Bestellung funktionieren? Enthält ein Boba Tea genau einen Drink und eine Sorte Boba? Eine Bestellung hat dann folgende Attribute: ID, Getränk, Boba, Datum, Bewertung?

Wir betrachten das "Bestellen" vereinfacht als die Erstellung/Kombination eines neuen Drinks. Ein neuer Drink besteht aus einer neuen Drink ID, einem Liquid, Boba, Datum und einem leeren Bewertungszustand. Wenn der drinkt erfolgreich erstellt wurde, kann er bewertet werden.

> Die Kurzbeschreibung von der Bewertung ist identisch mit der zum Abrufen einer Bestellung. Ist das richtig so? Ist eine Bewertung im Zahlenbereich zwischen 0 und 5 sowie einer Anzeige bis auf eine Nachkommastelle ok?

Da ist uns einer kleiner Fehler unterlaufen, wir werden die ii. "Retrieve Boba Tea Orders" entferenen. Der Zahlenbereich von 0 bis 5 ist okey. Die eine Kommastelle ebenfalls.

> Wie soll die Trennung zwischen Mitarbeiter und Kunde erfolgen? Genügt es die Funktionen zur Organisation von Drinks und Bobas über ein Passwort zu sichern?

Es reicht wenn es zwei verschiedene "Bereiche" gibt. Der Mitarbeiter Bereich soll einem universellen Passwort geschützt sind. 

> Gibt es ein oder zwei bevorzugte Farben für das Design der Website?

Geld/Rot/Orange

Danke für die Rückmeldung und die Fragen!
Tobias und Fabian

**Comment by bohgman** [comment]
Danke für die Erläuterungen!
Ist das Klassenmodel so in Ordnung für euch?

![Domain Model](https://user-images.githubusercontent.com/44813766/123148032-0a519600-d460-11eb-88e0-99108af96ed4.png)


Beste Grüße

**Comment by AlistarMidCarry** [comment]
Ja, das Model sieht gut aus soweit. Man könnte noch die Relationen ergänzen. Theoretisch kann man in jeden Tea beliebig viele Drinks und Toppings reinmischen.

**Comment by bohgman** [comment]
Ok die relation habe ich hinzugefügt. Aus Fabians Antort habe ich entnommen, dass ein Tee aus einem Getränk und Boba besteht. Daher habe ich das Disign Model erstmal so angelegt. Im agilen Entwicklungsprozess können ja noch Anpassungen erfolgen.

**Comment by bohgman** [comment]
![Design Model](https://user-images.githubusercontent.com/44813766/123148404-7b914900-d460-11eb-8b15-60d84c7f7288.png)

**Comment by AlistarMidCarry** [comment]
Das Model ist soweit ausreichend, danke dir!

**Comment by bohgman** [comment]
FYI:
https://bubbleboba-a0991.web.app/

Die erste Klasse Drink (ohne Validierung) ist online.
